# Exploring KRAS Expression

## Motivation

In the paper ["Tumor RAS Gene Expression Levels Are Influenced by the Mutational Status of RAS Genes and Both Upstream and Downstream RAS Pathway Genes"](https://journals.sagepub.com/doi/pdf/10.1177/1176935117711944), the authors studied relationship between RAS gene mutational status and messenger RNA expression. They saw higher levels of KRAS expression for samples that have KRAS mutation relative to samples without KRAS mutation for several cancer subtypes. The analysis was conducted using patient data from the [The Cancer Genome Atlas (TCGA)](https://www.cancer.gov/ccg/research/genome-sequencing/tcga) project, and we are curious whether similar trends hold via cell line models.

![KRAS expression is elevated in KRAS-mutant samples from lung, pancreatic, and colon adenocarcinomas relative to WT samples.](https://raw.githubusercontent.com/fhdsl/S1_Intro_to_R/main/images/kras.png){width="450"}

The cell line models we use is from the [Dependency Map Project](https://depmap.org/portal/home/) (DepMap), where over a thousand cancer cell lines were profiled for various genomic features, including mutational status and RNA expression.

You will write code to recreate this figure using the DepMap data. Below are some suggestions on how to analyze it, but you can do it your own way!

Clears the environment.

```{r}
rm(list = ls())
```

## Analysis 1

### Load analysis package and DepMap data in

```{r, message=FALSE}
library(tidyverse)
library(esquisse)
library(knitr)
load(url("https://github.com/fhdsl/S1_Intro_to_R/raw/main/classroom_data/CCLE.RData"))
```

### Examine the number of cell lines profiled for `metadata`, `expression`, `mutation`

```{r}

```

### Examine the frequency of cancer subtypes in `metadata`

```{r}
kable(table(metadata$OncotreePrimaryDisease))
```

### Alternatively, you could create a barplot to show the frequency of cancer subtypes in `metadata`:

```{r}

```

Need to display the text more clearly? Consider [this discussion on StackOverflow](https://stackoverflow.com/questions/1330989/rotating-and-spacing-axis-labels-in-ggplot2).

### Filter `metadata` so that it contains the cancer subtype want to analyze.

```{r}
           
```

### Join `metadata`, `expression`, and `mutation` together.

```{r}

```

### Select the columns you want to analyze.

```{r}

```

### Create the boxplot figure!

```{r}

```

### Consider a different way to visualize this data: Create a histogram of KRAS expression for non-KRAS mutated cell lines, and a different histogram of KRAS expression for KRAS mutated cell lines.

```{r}

```

## Analysis 2

Three well-established genes that are activated by KRAS are PI3K, RAF, and RAL. We will use correlation analysis to see how the expression of KRAS, PI3K, RAF, and RAL are related linearly.

### Create scatterplots of `KRAS` expression vs. `PIK3CA`, `RAF1`, and `RALB`. What does the trend look like?

```{r}

```

### We can measure the correlation of two continuous variables via the `cor` function. What's the correlation of `KRAS` vs. `RAF1`?

```{r}
#for example,
#cor(c(1, 2, 3), c(-1, 2, 0))
#cor(dataframe$colA, dataframe$colB)
```

### Can you put the correlation in the title of your scatterplot?

```{r}

```
